sci-libs/gdal/gdal1-1.11.5.recipe

```diff
@@ -36,7 +36,11 @@ PATCHES="gdal1-$portVersion.patchset"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
-if [ $targetArchitecture != x86_gcc2 ]; then
+
+if [ "$targetArchitecture" != x86_gcc2 ]; then
+	commandSuffix=$secondaryArchSuffix
```

Comment:
This is maybe standard now, but $commandSuffix ends up always empty, there is not much point in adding cruft.

Reply:
It's empty in most cases, but not if you have a hybrid x86_64 with x86 second arch. On such a system `gdal1_x86` should provide e.g. `cmd:gdal_contour_x86` instead of `cmd:gdal_contour` because the latter is already provided by `gdal1`.
We don't want  `cmd:gdal_contour` to be provided by both the primary and secondary arch builds on x86_64/x86 hybrid.

Reply:
gdal1_x86 on x86_64 won't provide the commands anyway, only the libraries. It won't provide cmd:gdal_contour_x86 at all.

Reply:
Maybe we need to make this a new standard then? Leaving out "commandSuffix=$secondaryArchSuffix"?

Reply:
> gdal1_x86 on x86_64 won't provide the commands anyway, only the libraries. It won't provide cmd:gdal_contour_x86 at all.

I'm affraid this is not true. With what we have **right now** in this PR the gdal1_x86 package on x86_64 hybrid will provide the `cmd:` items.

That said, we ~could~ **_should probably_** change `INSTALL` this way:
```diff
 INSTALL()
 {
 	make install
 
 	prepareInstalledDevelLib libgdal
 	fixPkgconfig

+	if [ -n "$secondaryArchSuffix" ]; then
+		mkdir -p "$binDir"
+		mv "$commandBinDir"/gdal-config "$binDir"/gdal-config
+	fi
 
 	# devel package
 	packageEntries devel \
 		"$binDir"/gdal-config \
 		"$developDir"
+
+	if [ -n "$commandSuffix" ]; then
+		rm -rf "$binDir"
+	fi
}
```
Remark: this would require a small change in PROVIDES`:
```
PROVIDES="
	gdal1$secondaryArchSuffix = $portVersionCompat
	lib:libgdal$secondaryArchSuffix = $libVersionCompat
	"
if [ -z "$commandSuffix" ]; then
PROVIDES+="
	cmd:gdal_contour
	cmd:gdal_grid
	cmd:gdal_rasterize
	cmd:gdal_translate
	cmd:gdaladdo
	cmd:gdalbuildvrt
	cmd:gdaldem
	cmd:gdalenhance
	cmd:gdalinfo
	cmd:gdallocationinfo
	cmd:gdalmanage
	cmd:gdalserver
	cmd:gdalsrsinfo
	cmd:gdaltindex
	cmd:gdaltransform
	cmd:gdalwarp
	cmd:nearblack
	cmd:ogr2ogr
	cmd:ogrinfo
	cmd:ogrlineref
	cmd:ogrtindex
	cmd:testepsg
	"
```
Hmm, after all, I think this is what we should do.

BTW, I've just noticed that `gdal-config` should go to `$binDir`, not to `$commandBinDir`. That's fixed in the code above .

Remark: `defineDebugInfoPackage` will also need to be changed if we apply the above suggestion.

Reply:
Still, you can drop $commandSuffix altogether.

Reply:
`$commandSuffix` is required here because we're installing the runtimes in  `$commandBinDir`.

Reply:
It is used as a flag. We set it to an empty string when we're building for any primary arch or for x86 secondary arch on x86_gcc2h. Later, in `INSTALL`, we use it to remove the runtimes:
```bash
	if [ -n "$commandSuffix" ]; then
		rm -rf "$binDir"
	fi
```

Reply:
Maybe better standardize a new name and have it defined default by haikuporter.

Reply:
Well, I'm not very fond of having haikuporter define another variable with some string depending on whether or not x86_gcc2 is marked as broken in `ARCHITECTURES`. HaikuPorter does not care about special architectures (e.g. x86_gcc2) and this is nice.

Even if we could modify many recipes to remove the occurrences of `$commandSuffix` in their `PROVIDES*` and call `rm -rf $binDir` in `INSTALL` when we are building on x86_64 hybrid for x86 secondary arch, there will still be recipes using `$commandSuffix` for good reasons. For example:
* `sed`
* `zstd`

So keeping the current tradition is imho better than choosing another name for that variable.

Reply:
I'm more for removing a pile of bash tests which ends up in every recipe with a command, given that most of the recipes now don't build with gcc2. I'm just surprised that everyone seems to be happy with the current status quo.

Reply:
`commandBinDir` (and `commandSuffix`)  where introduced a long time ago for very good reasons. At that time most recipes only needed the `commandBinDir`.
A few months ago x86_64 hybrid arrived. BTW, this is great news and we all know **_you_** did it!
So, to keep our recipes correct for all possible combinations of primary arch vs secondary arch we had to add a `commandSuffix` variable for those recipes which were missing it but were using the `commandBinDir` variable. (I did this while you were working on x86_64 hybrid.)

To summarize,  for those recipes which do not build for x86_gcc2 primary arch we have two solutions when we build them for x86 secondary arch on x86_64 hybrid:
1) We can install the runtimes in `$binDir`, i.e. in `system/bin/x86/`. That's what what we've been doing until now.
2) We can decide to drop the runtimes. This is what we're doing in this PR.
3) Same as 1 but we move the runtimes to a subpackage. (That's what I did today for rhash in 1d16ba1414779f6e5af950a4601161d37779bf82.)

Keeping the name `commandSuffix` has these advantages:
* We can switch from any of the above solutions to any other solution with less git noise than if we used another name for that variable.
* The recipes include the same 6-lines code snippet which we are used to since a long time.
* All our current contributors already know how `commandBinDir` and `commandSuffix` work, and I have already documented this in the wiki: [Using the commandBinDir User Variable](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#using-the-commandbindir-user-variable)

That's why everyone is happy!

Reply:
Ok, I think these comments are fine, but should be done in private or through emails on the list, all these comments make it hard for people to understand what way to go, if so, could the comments be only for the benefit of the recipe?

Reply:
No problem, sometimes it good to have conversations about it, but I got told before (even years back) that  it doesn't solve the issue at hand atm ;) (but they shouldn't take too long) :)